components/InvoiceFinancing/index.tsx

```diff
@@ -324,6 +326,93 @@ function ApiInvoices() {
     })
   }
 
+  const financeInvoiceAutoSubmit = async () => {
+    if (!issuedProduct) {
+      toast.error('No issued product found')
+      return
+    }
+
+    if (selectedInvoiceIds.size === 0) {
+      toast.error('Please select at least one invoice')
+      return
+    }
+
+    const invoicesToFinance: PlatformInvoice[] = [...selectedInvoiceIds]
+      .map((invoiceId) =>
+        allPersistedInvoices.find(
+          (persistedInvoice) => persistedInvoice.id === invoiceId,
+        ),
+      )
+      .filter(Boolean) as PlatformInvoice[]
+
+    if (invoicesToFinance.length === 0) {
+      toast.error('Selected invoices not found')
+      return
+    }
+
+    // Validate required fields
+    const invalidInvoices = invoicesToFinance.filter((invoice) => {
+      return (
+        _.isNil(invoice.payorType) ||
+        _.isNil(invoice.dueDateIsoDate) ||
+        _.isEmpty(invoice.items)
+      )
+    })
+
+    if (invalidInvoices.length > 0) {
+      toast.error(
+        'Cannot submit these invoices because some fields are missing.',
+      )
+      return
+    }
+
+    const payload: FinanceInvoicePayload = {
+      invoices: invoicesToFinance,
+      issuedProductId: issuedProduct.id,
+      platformBusinessId: issuedProduct.platformBusinessId as string,
+    }
+
+    try {
+      const resp = await axiosWithApiKey(apiKey).post<{
+        invoices: Invoice[]
+      }>('/api/finance_invoice', payload)
+
+      // Update invoices with kanmonInvoiceId to show "Financed" status
+      resp.data.invoices.forEach((financedInvoice) => {
```

Comment:
you could have partial failures here. e.g. 2 out of 3 invoices succeeded and the 3rd failed.

IMO it'd be simpler to just let them finance one at a time? and front end blocks if they select more than one.

wdyt?